modules/standard/JSON.chpl

```diff
@@ -96,85 +179,185 @@ module JSON {
-      writer.writeLiteral("{");
-      return new AggregateSerializer(writer, _ending="}");
+    /*
+      Start serializing a list by writing the character ``[``.
+
+      :arg writer: The ``fileWriter`` to be used when serializing.
+      :arg size: The number of elements in the list.
+
+      :returns: A new ListSerializer
+    */
+    proc startList(writer: jsonWriter, size: int) throws {
+      writer.writeLiteral("[");
+      return new ListSerializer(writer);
     }
 
+    /*
+      Returned by ``startList`` or ``startTuple`` to provide the API for
+      serializing JSON lists.
+
+      A list will be serialized as a comma-separated series of serialized
+      elements between two square brackets. For example, serializing a list
+      with elements ``1``, ``2``, and ``3`` will produce the text:
+
+      .. code-block:: text
+
+        [1, 2, 3]
+
+      Empty lists will be serialized as ``[]``.
+    */
     record ListSerializer {
+      @chpldoc.nodoc
       var writer;
+      @chpldoc.nodoc
       var _first : bool = true;
 
+      /*
+        Serialize ``element``.
+
+        Writes a leading comma before serializing the element if this is not the
+        first element in the list.
+      */
       proc ref writeElement(const element: ?) throws {
         if !_first then writer._writeLiteral(", ");
         else _first = false;
 
         writer.write(element);
       }
 
-      @chpldoc.nodoc
+      /*
+        Ends serialization of the current list by writing the character ``]``.
+      */
       proc endList() throws {
         writer.writeLiteral("]");
       }
 
-      @chpldoc.nodoc
+      /*
+        Ends serialization of the current tuple by writing the character ``]``.
+      */
       proc endTuple() throws {
         endList();
       }
     }
 
-    @chpldoc.nodoc
-    proc startTuple(writer: jsonWriter, size: int) throws {
-      writer.writeLiteral("[");
-      return new ListSerializer(writer);
-    }
+    /*
+      Start serializing an array.
 
-    @chpldoc.nodoc
-    proc startList(writer: jsonWriter, size: int) throws {
-      writer.writeLiteral("[");
-      return new ListSerializer(writer);
+      :arg writer: The ``fileWriter`` to be used when serializing.
+      :arg size: The number of elements in the array.
+
+      :returns: A new ArraySerializer
+    */
+    proc startArray(writer: jsonWriter, size: int) throws {
+      return new ArraySerializer(writer);
     }
 
+    /*
+      Returned by ``startArray`` to provide the API for serializing arrays.
+
+      In this format, a 1D array will be serialized as a JSON list. For
+      example, an array with three values would be serialized as:
+
+      .. code-block:: text
+
+        [1, 2, 3]
+
+      Multidimensional arrays will be serialized as JSON lists of lists:
+
+      .. code-block:: text
+
+        [
+         [0, 1, 2],
+         [3, 4, 5],
+         [6, 7, 8]
+        ]
+
+      The indenting of these multidimensional arrays is not considered stable.
```

Comment:
Maybe put this in a warning block.

Reply:
I was doing that initially, but unfortunately there's a chpldoc bug where the warning overflows and includes all the type's methods. The same goes for the code block preceding this, so having the text here is blocking that issue as well.